libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/RequireMapboxNavigationDelegate.kt

```diff
@@ -0,0 +1,135 @@
+ * [MapboxNavigation] instance can be re-created with [MapboxNavigationApp.disable], or if all
+ * [MapboxNavigationApp.attach] lifecycles are destroyed.
+ *
+ * @param onCreatedObserver registered to the [Lifecycle.State.CREATED] lifecycle
+ * @param onStartedObserver registered to the [Lifecycle.State.STARTED] lifecycle
+ * @param onResumedObserver registered to the [Lifecycle.State.RESUMED] lifecycle
+ * @param onInitialize called when the property is read for the first time
+ */
+class RequireMapboxNavigationDelegate(
+    private val onCreatedObserver: MapboxNavigationObserver? = null,
+    private val onStartedObserver: MapboxNavigationObserver? = null,
+    private val onResumedObserver: MapboxNavigationObserver? = null,
+    private val onInitialize: (() -> Unit)? = null
+) : ReadOnlyProperty<LifecycleOwner, MapboxNavigation> {
+
+    private lateinit var lifecycleOwner: LifecycleOwner
+
+    private val lifecycleObserver = object : DefaultLifecycleObserver {
+        override fun onCreate(owner: LifecycleOwner) {
+            onCreatedObserver?.let { MapboxNavigationApp.registerObserver(it) }
+        }
+
+        override fun onDestroy(owner: LifecycleOwner) {
+            onCreatedObserver?.let { MapboxNavigationApp.unregisterObserver(it) }
+        }
+
+        override fun onStart(owner: LifecycleOwner) {
+            onStartedObserver?.let { MapboxNavigationApp.registerObserver(it) }
+        }
+
+        override fun onStop(owner: LifecycleOwner) {
+            onStartedObserver?.let { MapboxNavigationApp.unregisterObserver(it) }
+        }
+
+        override fun onResume(owner: LifecycleOwner) {
+            onResumedObserver?.let { MapboxNavigationApp.registerObserver(it) }
+        }
+
+        override fun onPause(owner: LifecycleOwner) {
+            onResumedObserver?.let { MapboxNavigationApp.unregisterObserver(it) }
+        }
+    }
+
+    /**
+     * Returns an instance of [MapboxNavigation], the first retrieval will attach the [thisRef]
+     * to [MapboxNavigationApp.attach]. If [MapboxNavigationApp.isSetup] is false after all
+     * observers and initializers, this property getter will crash.
+     *
+     * @param thisRef - the [LifecycleOwner] that needs access to [MapboxNavigation].
+     * @param property - ignored
+     */
+    override fun getValue(thisRef: LifecycleOwner, property: KProperty<*>): MapboxNavigation {
+        if (!this::lifecycleOwner.isInitialized) {
+            onInitialize?.invoke()
+            this.lifecycleOwner = thisRef
+            MapboxNavigationApp.attach(lifecycleOwner)
+            lifecycleOwner.lifecycle.addObserver(lifecycleObserver)
+        }
+        val mapboxNavigation = MapboxNavigationApp.current()
+        checkNotNull(mapboxNavigation) {
+            "MapboxNavigation cannot be null. Ensure that MapboxNavigationApp is setup and an" +
+                " attached lifecycle is at least CREATED."
```

Comment:
If I understand correctly, `current()` returns non-null only after `STARTED`: https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationOwner.kt#L28

Reply:
Not quite, the `carAppLifecycleObserver` is never `DESTROYED` but it will start and stop.

That is because the destroy state is terminal, it cannot be re-created after it is destoryed. But the Activity/Fragment can be created and destroyed. 

An activity must be `CREATED` in order to move the CarAppLifecycle to `STARTED`